tensorflow/c/kernels.cc

```diff
@@ -260,3 +268,24 @@ TF_Tensor* TF_AllocateOutput(TF_OpKernelContext* context, int index,
   }
   return result;
 }
+
+TF_Tensor* TF_AllocateTemp(TF_OpKernelContext* context, TF_DataType dtype, 
+                     int64_t* dims, int num_dims, TF_Status* status){
+  auto* cc_ctx = reinterpret_cast<::tensorflow::OpKernelContext*>(context);
+  TF_SetStatus(status, TF_OK, ""); 
+  tensorflow::TensorShape shape;
+  for(int i = 0; i < num_dims; ++i){
+    shape.AddDim(dims[i]); 
+  }
+  tensorflow::Status s;
+  tensorflow::Tensor tensor_temp;  
+  TF_Tensor* tf_tensor_temp; 
+  s = cc_ctx->allocate_temp(static_cast<tensorflow::DataType>(dtype), shape, &tensor_temp);
+  if (s.ok()){ 
```

Comment:
I find it easier to reason about the code when status checkers check for error (instead of success) and return right away.
Something like
```cpp
if (!s.ok()) {
  ::tensorflow::Set_TF_Status_from_Status(status, s); 
  return nullptr;
}
tf_tensor_temp = TF_TensorFromTensor(tensor_temp, &s);
if (!s.ok()) {
  ::tensorflow::Set_TF_Status_from_Status(status, s); 
  return nullptr;
}
... success path ....
```